### PR TITLE
shouldPausePlacement short circuits when _forceFullPlacement

### PR DESCRIPTION
**Pull request by schlosna**
## Launch Checklist

When rendering many symbols on a map, there is a lot of time spent iterating through `continuePlacement()` and callbacks to `shouldPausePlacement()`. The current implementation of `shouldPausePlacement()` always invokes `browser.now()` to compute the elapsed time, even when `_forceFullPlacement == true`.

Given this can be a hot path when there are many symbols, we can check `_forceFullPlacement` and potentially short circuit without needing to compute the elapsed time, saving roughly ~30ns to ~50ns per invocation.

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [x] Briefly describe the changes in this PR.
 - [x] Link to related issues.
     - None that I could find
 - [x] Include before/after visuals or gifs if this PR includes visual changes.
     - N/A
 - [ ] Write tests for all new functionality.
 - [x] Document any changes to public APIs.
     - N/A
 - [x] Post benchmark scores.
 - [x] Add an entry to `CHANGELOG.md` under the `## main` section.

### CODE REVIEW

**Comment by HarelM** [comment]
Thanks for taking the time to open this PR!
Can you add a changelog entry for this change as well?
Can you think of a test to cover this change?
Do you have a way to make sure this test actually improve performance? I see the code change and I understand why this is better, and the code itself isn't bad, so this change is obviously welcome, just want to make sure we are actually making a difference...

File: src/style/pauseable_placement.ts [hunk 1]
```diff
@@ -97,8 +97,10 @@ class PauseablePlacement {
         const startTime = browser.now();
 
         const shouldPausePlacement = () => {
-            const elapsedTime = browser.now() - startTime;
-            return this._forceFullPlacement ? false : elapsedTime > 2;
+            if (this._forceFullPlacement) {
+                return false;
+            }
+            return browser.now() - startTime > 2;
```
Comment (schlosna) [inline src/style/pauseable_placement.ts:103]: I'm not sure what style you all prefer, so curious if you would prefer ternary for conciseness?

```suggestion
            return this._forceFullPlacement ? false : (browser.now() - startTime) > 2;
```

**Review by schlosna — COMMENTED** [review]
Thanks for reviewing! I have updated the changelog. I was finally able to get the benchmarks running locally and capture output for the ones that seem most relevant to this change around placement and symbol layout and attached below. The improvement does not seem statistically conclusive, but there are slight improvements in (20% trimmed) Mean compared to current `main` bc6d6f4.

I'm not super familiar with the maplibre-gl-js codebase, so if you have pointers on specific tests that should exercise the (pausable) placement bits, I can try to add some coverage there.

image

image

image

image

**Comment by schlosna** [comment]
Hi @HarelM , I just wanted to check to see if there were any further comments, questions, or changes you'd like to see for this PR. Thanks!

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
I'll wait for the tests to finish and merge. Thanks!

**Comment by schlosna** [comment]
Excellent, thanks!

**Pull request merged**